networkit/exceptions.py

```diff
@@ -0,0 +1,3 @@
+class ReducedFunctionality(Warning):
```

Comment:
I would prefer to rename this class to `ReducedFunctionalityWarning`. `exceptions.ReducedFunctionality` otherwise suggests that this is a (more critical) exception. This would also be in line with the `Warning` subclasses of the Python standard library.

Reply:
Makes sense to me.